openapi-validation-api/src/main/java/com/getyourguide/openapi/validation/api/selector/DefaultTrafficSelector.java

```diff
@@ -65,6 +71,17 @@ public boolean shouldFailOnResponseViolation(RequestMetaData request) {
     }
 
     private boolean isExcludedRequest(RequestMetaData request) {
+        return isRequestExcludedByHeader(request) || isRequestExcludedByPath(request);
+    }
+
+    private boolean isRequestExcludedByHeader(RequestMetaData request) {
+        return excludedHeaders.stream().anyMatch(excludedHeader -> {
+            var headerValue = request.getHeaders().get(excludedHeader.headerName());
+            return headerValue != null && excludedHeader.headerValuePattern().matcher(headerValue).matches();
```

Comment:
Just out of curiosity, is it possible to have headerValue == null IYO?

Reply:
Yes it is possible because we go through the list of headers one wants to exclude like `x-is-bot`. Then `request.getHeaders().get("x-is-bot");` can return null if there is no such header sent in the request.

Reply:
<img width="866" alt="Screenshot 2023-05-26 at 08 48 59" src="https://github.com/getyourguide/openapi-validation-java/assets/398400/f067693f-de7a-4275-91bb-65dfd5fa67eb">